### PR TITLE
support jdbc for mysql 8.0 (#314)

### LINKED ISSUE #314: Bug: when java jdbc connect arana, error: Unknown system variable 'query_cache_size'
ref #272 

**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

java client error:

```
/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/bin/java -javaagent:/Applications/IntelliJ IDEA.app/Contents/lib/idea_rt.jar=56369:/Applications/IntelliJ IDEA.app/Contents/bin -Dfile.encoding=UTF-8 -classpath /Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/charsets.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/deploy.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/ext/cldrdata.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/ext/dnsns.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/ext/jaccess.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/ext/jfxrt.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/ext/localedata.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/ext/nashorn.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/ext/sunec.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/ext/sunjce_provider.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/ext/sunpkcs11.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/ext/zipfs.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/javaws.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/jce.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/jfr.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/jfxswt.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/jsse.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/management-agent.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/plugin.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/resources.jar:/Library/Java/JavaVirtualMachines/jdk1.8.0_161.jdk/Contents/Home/jre/lib/rt.jar:/Users/dongzonglei/source_code/Github/java-book-code/java-book-self/target/classes:/Users/dongzonglei/server/repository/com/google/guava/guava/27.1-jre/guava-27.1-jre.jar:/Users/dongzonglei/server/repository/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar:/Users/dongzonglei/server/repository/com/google/guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar:/Users/dongzonglei/server/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar:/Users/dongzonglei/server/repository/org/checkerframework/checker-qual/2.5.2/checker-qual-2.5.2.jar:/Users/dongzonglei/server/repository/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.jar:/Users/dongzonglei/server/repository/org/codehaus/mojo/animal-sniffer-annotations/1.17/animal-sniffer-annotations-1.17.jar:/Users/dongzonglei/server/repository/com/alibaba/transmittable-thread-local/2.10.2/transmittable-thread-local-2.10.2.jar:/Users/dongzonglei/server/repository/com/alibaba/easyexcel/1.1.2-beta5/easyexcel-1.1.2-beta5.jar:/Users/dongzonglei/server/repository/org/apache/poi/poi/3.17/poi-3.17.jar:/Users/dongzonglei/server/repository/org/apache/commons/commons-collections4/4.1/commons-collections4-4.1.jar:/Users/dongzonglei/server/repository/org/apache/poi/poi-ooxml/3.17/poi-ooxml-3.17.jar:/Users/dongzonglei/server/repository/org/apache/poi/poi-ooxml-schemas/3.17/poi-ooxml-schemas-3.17.jar:/Users/dongzonglei/server/repository/org/apache/xmlbeans/xmlbeans/2.6.0/xmlbeans-2.6.0.jar:/Users/dongzonglei/server/repository/stax/stax-api/1.0.1/stax-api-1.0.1.jar:/Users/dongzonglei/server/repository/com/github/virtuald/curvesapi/1.04/curvesapi-1.04.jar:/Users/dongzonglei/server/repository/com/lmax/disruptor/3.4.2/disruptor-3.4.2.jar:/Users/dongzonglei/server/repository/com/zaxxer/HikariCP/3.3.1/HikariCP-3.3.1.jar:/Users/dongzonglei/server/repository/org/slf4j/slf4j-api/1.7.25/slf4j-api-1.7.25.jar:/Users/dongzonglei/server/repository/com/typesafe/akka/akka-actor_2.11/2.3.7/akka-actor_2.11-2.3.7.jar:/Users/dongzonglei/server/repository/org/scala-lang/scala-library/2.11.4/scala-library-2.11.4.jar:/Users/dongzonglei/server/repository/com/typesafe/config/1.2.1/config-1.2.1.jar:/Users/dongzonglei/server/repository/com/typesafe/akka/akka-agent_2.11/2.3.9/akka-agent_2.11-2.3.9.jar:/Users/dongzonglei/server/repository/org/scala-stm/scala-stm_2.11/0.7/scala-stm_2.11-0.7.jar:/Users/dongzonglei/server/repository/com/typesafe/akka/akka-transactor_2.11/2.3.8/akka-transactor_2.11-2.3.8.jar:/Users/dongzonglei/server/repository/org/apache/zookeeper/zookeeper/3.4.6/zookeeper-3.4.6.jar:/Users/dongzonglei/server/repository/org/slf4j/slf4j-log4j12/1.6.1/slf4j-log4j12-1.6.1.jar:/Users/dongzonglei/server/repository/log4j/log4j/1.2.16/log4j-1.2.16.jar:/Users/dongzonglei/server/repository/jline/jline/0.9.94/jline-0.9.94.jar:/Users/dongzonglei/server/repository/io/netty/netty/3.7.0.Final/netty-3.7.0.Final.jar:/Users/dongzonglei/server/repository/org/apache/curator/curator-framework/4.0.0/curator-framework-4.0.0.jar:/Users/dongzonglei/server/repository/org/apache/curator/curator-client/4.0.0/curator-client-4.0.0.jar:/Users/dongzonglei/server/repository/org/apache/curator/curator-recipes/4.0.0/curator-recipes-4.0.0.jar:/Users/dongzonglei/server/repository/javax/servlet/javax.servlet-api/3.1.0/javax.servlet-api-3.1.0.jar:/Users/dongzonglei/server/repository/com/fasterxml/jackson/core/jackson-databind/2.6.6/jackson-databind-2.6.6.jar:/Users/dongzonglei/server/repository/com/fasterxml/jackson/core/jackson-core/2.6.6/jackson-core-2.6.6.jar:/Users/dongzonglei/server/repository/com/fasterxml/jackson/core/jackson-annotations/2.6.6/jackson-annotations-2.6.6.jar:/Users/dongzonglei/server/repository/org/codehaus/jackson/jackson-mapper-asl/1.9.13/jackson-mapper-asl-1.9.13.jar:/Users/dongzonglei/server/repository/org/codehaus/jackson/jackson-core-asl/1.9.13/jackson-core-asl-1.9.13.jar:/Users/dongzonglei/server/repository/org/postgresql/postgresql/42.2.8/postgresql-42.2.8.jar:/Users/dongzonglei/server/repository/org/redisson/redisson/3.11.4/redisson-3.11.4.jar:/Users/dongzonglei/server/repository/io/netty/netty-common/4.1.41.Final/netty-common-4.1.41.Final.jar:/Users/dongzonglei/server/repository/io/netty/netty-codec/4.1.41.Final/netty-codec-4.1.41.Final.jar:/Users/dongzonglei/server/repository/io/netty/netty-buffer/4.1.41.Final/netty-buffer-4.1.41.Final.jar:/Users/dongzonglei/server/repository/io/netty/netty-transport/4.1.41.Final/netty-transport-4.1.41.Final.jar:/Users/dongzonglei/server/repository/io/netty/netty-resolver/4.1.41.Final/netty-resolver-4.1.41.Final.jar:/Users/dongzonglei/server/repository/io/netty/netty-resolver-dns/4.1.41.Final/netty-resolver-dns-4.1.41.Final.jar:/Users/dongzonglei/server/repository/io/netty/netty-codec-dns/4.1.41.Final/netty-codec-dns-4.1.41.Final.jar:/Users/dongzonglei/server/repository/io/netty/netty-handler/4.1.41.Final/netty-handler-4.1.41.Final.jar:/Users/dongzonglei/server/repository/javax/cache/cache-api/1.0.0/cache-api-1.0.0.jar:/Users/dongzonglei/server/repository/io/projectreactor/reactor-core/3.2.6.RELEASE/reactor-core-3.2.6.RELEASE.jar:/Users/dongzonglei/server/repository/org/reactivestreams/reactive-streams/1.0.2/reactive-streams-1.0.2.jar:/Users/dongzonglei/server/repository/io/reactivex/rxjava2/rxjava/2.2.7/rxjava-2.2.7.jar:/Users/dongzonglei/server/repository/de/ruedigermoeller/fst/2.57/fst-2.57.jar:/Users/dongzonglei/server/repository/org/objenesis/objenesis/2.5.1/objenesis-2.5.1.jar:/Users/dongzonglei/server/repository/com/fasterxml/jackson/dataformat/jackson-dataformat-yaml/2.9.10/jackson-dataformat-yaml-2.9.10.jar:/Users/dongzonglei/server/repository/org/yaml/snakeyaml/1.23/snakeyaml-1.23.jar:/Users/dongzonglei/server/repository/net/bytebuddy/byte-buddy/1.9.16/byte-buddy-1.9.16.jar:/Users/dongzonglei/server/repository/org/jodd/jodd-bean/5.0.13/jodd-bean-5.0.13.jar:/Users/dongzonglei/server/repository/org/jodd/jodd-core/5.0.13/jodd-core-5.0.13.jar:/Users/dongzonglei/server/repository/de/bwaldvogel/mongo-java-server/1.21.0/mongo-java-server-1.21.0.jar:/Users/dongzonglei/server/repository/de/bwaldvogel/mongo-java-server-core/1.21.0/mongo-java-server-core-1.21.0.jar:/Users/dongzonglei/server/repository/de/bwaldvogel/mongo-java-server-memory-backend/1.21.0/mongo-java-server-memory-backend-1.21.0.jar:/Users/dongzonglei/server/repository/org/mongodb/mongo-java-driver/3.4.3/mongo-java-driver-3.4.3.jar:/Users/dongzonglei/server/repository/io/etcd/jetcd-core/0.4.1/jetcd-core-0.4.1.jar:/Users/dongzonglei/server/repository/io/etcd/jetcd-common/0.4.1/jetcd-common-0.4.1.jar:/Users/dongzonglei/server/repository/io/etcd/jetcd-resolver/0.4.1/jetcd-resolver-0.4.1.jar:/Users/dongzonglei/server/repository/io/grpc/grpc-core/1.24.0/grpc-core-1.24.0.jar:/Users/dongzonglei/server/repository/io/grpc/grpc-api/1.24.0/grpc-api-1.24.0.jar:/Users/dongzonglei/server/repository/io/grpc/grpc-context/1.24.0/grpc-context-1.24.0.jar:/Users/dongzonglei/server/repository/com/google/errorprone/error_prone_annotations/2.3.2/error_prone_annotations-2.3.2.jar:/Users/dongzonglei/server/repository/com/google/android/annotations/4.1.1.4/annotations-4.1.1.4.jar:/Users/dongzonglei/server/repository/io/perfmark/perfmark-api/0.17.0/perfmark-api-0.17.0.jar:/Users/dongzonglei/server/repository/io/opencensus/opencensus-api/0.21.0/opencensus-api-0.21.0.jar:/Users/dongzonglei/server/repository/io/opencensus/opencensus-contrib-grpc-metrics/0.21.0/opencensus-contrib-grpc-metrics-0.21.0.jar:/Users/dongzonglei/server/repository/io/grpc/grpc-netty/1.24.0/grpc-netty-1.24.0.jar:/Users/dongzonglei/server/repository/io/netty/netty-codec-http2/4.1.38.Final/netty-codec-http2-4.1.38.Final.jar:/Users/dongzonglei/server/repository/io/netty/netty-codec-http/4.1.38.Final/netty-codec-http-4.1.38.Final.jar:/Users/dongzonglei/server/repository/io/netty/netty-handler-proxy/4.1.38.Final/netty-handler-proxy-4.1.38.Final.jar:/Users/dongzonglei/server/repository/io/netty/netty-codec-socks/4.1.38.Final/netty-codec-socks-4.1.38.Final.jar:/Users/dongzonglei/server/repository/io/grpc/grpc-protobuf/1.24.0/grpc-protobuf-1.24.0.jar:/Users/dongzonglei/server/repository/com/google/protobuf/protobuf-java/3.9.0/protobuf-java-3.9.0.jar:/Users/dongzonglei/server/repository/com/google/api/grpc/proto-google-common-protos/1.12.0/proto-google-common-protos-1.12.0.jar:/Users/dongzonglei/server/repository/io/grpc/grpc-protobuf-lite/1.24.0/grpc-protobuf-lite-1.24.0.jar:/Users/dongzonglei/server/repository/io/grpc/grpc-stub/1.24.0/grpc-stub-1.24.0.jar:/Users/dongzonglei/server/repository/io/grpc/grpc-grpclb/1.24.0/grpc-grpclb-1.24.0.jar:/Users/dongzonglei/server/repository/com/google/protobuf/protobuf-java-util/3.9.0/protobuf-java-util-3.9.0.jar:/Users/dongzonglei/server/repository/net/jodah/failsafe/2.3.1/failsafe-2.3.1.jar:/Users/dongzonglei/server/repository/com/alibaba/nacos/nacos-client/1.1.4/nacos-client-1.1.4.jar:/Users/dongzonglei/server/repository/com/alibaba/nacos/nacos-common/1.1.4/nacos-common-1.1.4.jar:/Users/dongzonglei/server/repository/com/alibaba/nacos/nacos-api/1.1.4/nacos-api-1.1.4.jar:/Users/dongzonglei/server/repository/com/alibaba/fastjson/1.2.58/fastjson-1.2.58.jar:/Users/dongzonglei/server/repository/commons-codec/commons-codec/1.11/commons-codec-1.11.jar:/Users/dongzonglei/server/repository/io/prometheus/simpleclient/0.5.0/simpleclient-0.5.0.jar:/Users/dongzonglei/server/repository/cglib/cglib/3.2.3/cglib-3.2.3.jar:/Users/dongzonglei/server/repository/org/ow2/asm/asm/5.0.4/asm-5.0.4.jar:/Users/dongzonglei/server/repository/org/apache/ant/ant/1.9.6/ant-1.9.6.jar:/Users/dongzonglei/server/repository/org/apache/ant/ant-launcher/1.9.6/ant-launcher-1.9.6.jar:/Users/dongzonglei/server/repository/org/openjdk/jol/jol-core/0.9/jol-core-0.9.jar:/Users/dongzonglei/server/repository/com/redislabs/jrebloom/1.2.0/jrebloom-1.2.0.jar:/Users/dongzonglei/server/repository/com/google/code/gson/gson/2.8.0/gson-2.8.0.jar:/Users/dongzonglei/server/repository/org/apache/commons/commons-pool2/2.4.2/commons-pool2-2.4.2.jar:/Users/dongzonglei/server/repository/junit/junit/4.12/junit-4.12.jar:/Users/dongzonglei/server/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar:/Users/dongzonglei/server/repository/redis/clients/jedis/3.6.2/jedis-3.6.2.jar:/Users/dongzonglei/server/repository/org/apache/lucene/lucene-core/4.0.0/lucene-core-4.0.0.jar:/Users/dongzonglei/server/repository/org/mp4parser/isoparser/1.9.41/isoparser-1.9.41.jar:/Users/dongzonglei/server/repository/org/bytedeco/ffmpeg/4.2.2-1.5.3/ffmpeg-4.2.2-1.5.3.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3.jar:/Users/dongzonglei/server/repository/org/bytedeco/ffmpeg-platform/4.2.2-1.5.3/ffmpeg-platform-4.2.2-1.5.3.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp-platform/1.5.3/javacpp-platform-1.5.3.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3-android-arm.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3-android-arm64.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3-android-x86.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3-android-x86_64.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3-ios-arm64.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3-ios-x86_64.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3-linux-armhf.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3-linux-arm64.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3-linux-ppc64le.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3-linux-x86.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3-linux-x86_64.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3-macosx-x86_64.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3-windows-x86.jar:/Users/dongzonglei/server/repository/org/bytedeco/javacpp/1.5.3/javacpp-1.5.3-windows-x86_64.jar:/Users/dongzonglei/server/repository/org/bytedeco/ffmpeg/4.2.2-1.5.3/ffmpeg-4.2.2-1.5.3-android-arm.jar:/Users/dongzonglei/server/repository/org/bytedeco/ffmpeg/4.2.2-1.5.3/ffmpeg-4.2.2-1.5.3-android-arm64.jar:/Users/dongzonglei/server/repository/org/bytedeco/ffmpeg/4.2.2-1.5.3/ffmpeg-4.2.2-1.5.3-android-x86.jar:/Users/dongzonglei/server/repository/org/bytedeco/ffmpeg/4.2.2-1.5.3/ffmpeg-4.2.2-1.5.3-android-x86_64.jar:/Users/dongzonglei/server/repository/org/bytedeco/ffmpeg/4.2.2-1.5.3/ffmpeg-4.2.2-1.5.3-linux-x86.jar:/Users/dongzonglei/server/repository/org/bytedeco/ffmpeg/4.2.2-1.5.3/ffmpeg-4.2.2-1.5.3-linux-x86_64.jar:/Users/dongzonglei/server/repository/org/bytedeco/ffmpeg/4.2.2-1.5.3/ffmpeg-4.2.2-1.5.3-linux-armhf.jar:/Users/dongzonglei/server/repository/org/bytedeco/ffmpeg/4.2.2-1.5.3/ffmpeg-4.2.2-1.5.3-linux-arm64.jar:/Users/dongzonglei/server/repository/org/bytedeco/ffmpeg/4.2.2-1.5.3/ffmpeg-4.2.2-1.5.3-linux-ppc64le.jar:/Users/dongzonglei/server/repository/org/bytedeco/ffmpeg/4.2.2-1.5.3/ffmpeg-4.2.2-1.5.3-macosx-x86_64.jar:/Users/dongzonglei/server/repository/org/bytedeco/ffmpeg/4.2.2-1.5.3/ffmpeg-4.2.2-1.5.3-windows-x86.jar:/Users/dongzonglei/server/repository/org/bytedeco/ffmpeg/4.2.2-1.5.3/ffmpeg-4.2.2-1.5.3-windows-x86_64.jar:/Users/dongzonglei/server/repository/org/msgpack/msgpack/0.6.12/msgpack-0.6.12.jar:/Users/dongzonglei/server/repository/com/googlecode/json-simple/json-simple/1.1.1/json-simple-1.1.1.jar:/Users/dongzonglei/server/repository/org/javassist/javassist/3.18.1-GA/javassist-3.18.1-GA.jar:/Users/dongzonglei/server/repository/org/apache/calcite/calcite-core/1.26.0/calcite-core-1.26.0.jar:/Users/dongzonglei/server/repository/org/apache/calcite/calcite-linq4j/1.26.0/calcite-linq4j-1.26.0.jar:/Users/dongzonglei/server/repository/org/apache/calcite/avatica/avatica-core/1.17.0/avatica-core-1.17.0.jar:/Users/dongzonglei/server/repository/org/apache/calcite/avatica/avatica-metrics/1.17.0/avatica-metrics-1.17.0.jar:/Users/dongzonglei/server/repository/org/apiguardian/apiguardian-api/1.1.0/apiguardian-api-1.1.0.jar:/Users/dongzonglei/server/repository/com/esri/geometry/esri-geometry-api/2.2.0/esri-geometry-api-2.2.0.jar:/Users/dongzonglei/server/repository/com/google/uzaygezen/uzaygezen-core/0.2/uzaygezen-core-0.2.jar:/Users/dongzonglei/server/repository/com/jayway/jsonpath/json-path/2.4.0/json-path-2.4.0.jar:/Users/dongzonglei/server/repository/net/minidev/json-smart/2.3/json-smart-2.3.jar:/Users/dongzonglei/server/repository/net/minidev/accessors-smart/1.2/accessors-smart-1.2.jar:/Users/dongzonglei/server/repository/com/yahoo/datasketches/sketches-core/0.9.0/sketches-core-0.9.0.jar:/Users/dongzonglei/server/repository/com/yahoo/datasketches/memory/0.9.0/memory-0.9.0.jar:/Users/dongzonglei/server/repository/net/hydromatic/aggdesigner-algorithm/6.0/aggdesigner-algorithm-6.0.jar:/Users/dongzonglei/server/repository/commons-lang/commons-lang/2.4/commons-lang-2.4.jar:/Users/dongzonglei/server/repository/org/apache/commons/commons-dbcp2/2.6.0/commons-dbcp2-2.6.0.jar:/Users/dongzonglei/server/repository/org/apache/commons/commons-lang3/3.8/commons-lang3-3.8.jar:/Users/dongzonglei/server/repository/org/codehaus/janino/commons-compiler/3.0.11/commons-compiler-3.0.11.jar:/Users/dongzonglei/server/repository/org/codehaus/janino/janino/3.0.11/janino-3.0.11.jar:/Users/dongzonglei/server/repository/org/apache/calcite/calcite-server/1.26.0/calcite-server-1.26.0.jar:/Users/dongzonglei/server/repository/commons-dbcp/commons-dbcp/1.4/commons-dbcp-1.4.jar:/Users/dongzonglei/server/repository/commons-pool/commons-pool/1.5.4/commons-pool-1.5.4.jar:/Users/dongzonglei/server/repository/mysql/mysql-connector-java/5.1.47/mysql-connector-java-5.1.47.jar:/Users/dongzonglei/server/repository/com/alibaba/druid/1.2.5/druid-1.2.5.jar:/Users/dongzonglei/server/repository/com/aliyun/aliyun-java-sdk-dataworks-public/3.3.1/aliyun-java-sdk-dataworks-public-3.3.1.jar:/Users/dongzonglei/server/repository/commons-io/commons-io/2.6/commons-io-2.6.jar:/Users/dongzonglei/server/repository/org/jsoup/jsoup/1.11.3/jsoup-1.11.3.jar:/Users/dongzonglei/server/repository/org/apache/httpcomponents/httpclient/4.5/httpclient-4.5.jar:/Users/dongzonglei/server/repository/commons-logging/commons-logging/1.2/commons-logging-1.2.jar:/Users/dongzonglei/server/repository/org/apache/httpcomponents/httpmime/4.5/httpmime-4.5.jar:/Users/dongzonglei/server/repository/org/apache/httpcomponents/httpcore/4.4.1/httpcore-4.4.1.jar:/Users/dongzonglei/server/repository/com/squareup/okhttp3/okhttp/3.4.1/okhttp-3.4.1.jar:/Users/dongzonglei/server/repository/com/squareup/okio/okio/1.9.0/okio-1.9.0.jar:/Users/dongzonglei/server/repository/org/testcontainers/testcontainers/1.16.3/testcontainers-1.16.3.jar:/Users/dongzonglei/server/repository/org/apache/commons/commons-compress/1.21/commons-compress-1.21.jar:/Users/dongzonglei/server/repository/org/rnorth/duct-tape/duct-tape/1.0.8/duct-tape-1.0.8.jar:/Users/dongzonglei/server/repository/org/jetbrains/annotations/17.0.0/annotations-17.0.0.jar:/Users/dongzonglei/server/repository/com/github/docker-java/docker-java-api/3.2.12/docker-java-api-3.2.12.jar:/Users/dongzonglei/server/repository/com/github/docker-java/docker-java-transport-zerodep/3.2.12/docker-java-transport-zerodep-3.2.12.jar:/Users/dongzonglei/server/repository/com/github/docker-java/docker-java-transport/3.2.12/docker-java-transport-3.2.12.jar:/Users/dongzonglei/server/repository/net/java/dev/jna/jna/5.8.0/jna-5.8.0.jar arana.TestInsert
Tue Jul 26 17:01:57 CST 2022 WARN: Establishing SSL connection without server's identity verification is not recommended. According to MySQL 5.5.45+, 5.6.26+ and 5.7.6+ requirements SSL connection must be established by default if explicit option isn't set. For compliance with existing applications not using SSL the verifyServerCertificate property is set to 'false'. You need either to explicitly disable SSL by setting useSSL=false, or set useSSL=true and provide truststore for server certificate verification.
Exception in thread "main" java.sql.SQLException: Unknown system variable 'query_cache_size'
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:965)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3978)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3914)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2530)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2683)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2491)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2449)
	at com.mysql.jdbc.StatementImpl.executeQuery(StatementImpl.java:1381)
	at com.mysql.jdbc.ConnectionImpl.loadServerVariables(ConnectionImpl.java:3797)
	at com.mysql.jdbc.ConnectionImpl.initializePropsFromServer(ConnectionImpl.java:3230)
	at com.mysql.jdbc.ConnectionImpl.connectOneTryOnly(ConnectionImpl.java:2243)
	at com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2025)
	at com.mysql.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:778)
	at com.mysql.jdbc.JDBC4Connection.<init>(JDBC4Connection.java:47)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
	at com.mysql.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:386)
	at com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:330)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at arana.TestInsert.main(TestInsert.java:17)

Process finished with exit code 1
```

arana server error:

```
2022-07-26T17:01:57.827+0800    DEBUG   executor/redirect.go:172        ComQuery: /* mysql-connector-java-5.1.47 ( Revision: fe1903b1ecb4a96a917f7ed3190d80c049b1de29 ) */SELECT  @@session.auto_increment_increment AS auto_increment_increment, @@character_set_client AS character_set_client, @@character_set_connection AS character_set_connection, @@character_set_results AS character_set_results, @@character_set_server AS character_set_server, @@collation_server AS collation_server, @@collation_connection AS collation_connection, @@init_connect AS init_connect, @@interactive_timeout AS interactive_timeout, @@license AS license, @@lower_case_table_names AS lower_case_table_names, @@max_allowed_packet AS max_allowed_packet, @@net_buffer_length AS net_buffer_length, @@net_write_timeout AS net_write_timeout, @@query_cache_size AS query_cache_size, @@query_cache_type AS query_cache_type, @@sql_mode AS sql_mode, @@system_time_zone AS system_time_zone, @@time_zone AS time_zone, @@tx_isolation AS transaction_isolation, @@wait_timeout AS wait_timeout
2022-07-26T17:01:57.827+0800    DEBUG   runtime/runtime.go:678  call upstream: db=, id=node0, sql="SELECT @@auto_increment_increment AS `auto_increment_increment`,@@character_set_client AS `character_set_client`,@@character_set_connection AS `character_set_connection`,@@character_set_results AS `character_set_results`,@@character_set_server AS `character_set_server`,@@collation_server AS `collation_server`,@@collation_connection AS `collation_connection`,@@init_connect AS `init_connect`,@@interactive_timeout AS `interactive_timeout`,@@license AS `license`,@@lower_case_table_names AS `lower_case_table_names`,@@max_allowed_packet AS `max_allowed_packet`,@@net_buffer_length AS `net_buffer_length`,@@net_write_timeout AS `net_write_timeout`,@@query_cache_size AS `query_cache_size`,@@query_cache_type AS `query_cache_type`,@@sql_mode AS `sql_mode`,@@system_time_zone AS `system_time_zone`,@@time_zone AS `time_zone`,@@tx_isolation AS `transaction_isolation`,@@wait_timeout AS `wait_timeout`", args=[]
2022-07-26T17:01:57.835+0800    ERROR   mysql/execute_handle.go:97      get dataset error 4: Unknown system variable 'query_cache_size' (errno 1193) (sqlstate HY000)
github.com/arana-db/arana/pkg/mysql.(*Listener).handleQuery
        /Users/dongzonglei/source_code/Github/arana/pkg/mysql/execute_handle.go:97
github.com/arana-db/arana/pkg/mysql.(*Listener).ExecuteCommand
        /Users/dongzonglei/source_code/Github/arana/pkg/mysql/server.go:553
github.com/arana-db/arana/pkg/mysql.(*Listener).handle
        /Users/dongzonglei/source_code/Github/arana/pkg/mysql/server.go:203
```

Test code, ref: #272 .

### PR DESCRIPTION
**Pull request by csynineyang**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #314 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by dongzl — APPROVED** [review]

**Review by georgehao — APPROVED** [review]

**Review by Lvnszn — APPROVED** [review]

File: pkg/runtime/plan/dml/simple_select.go [hunk 1]
```diff
@@ -70,6 +72,19 @@ func (s *SimpleQueryPlan) ExecIn(ctx context.Context, conn proto.VConn) (proto.R
...
+		args_cachesize_57 := "@@query_cache_size"
+		args_cachesize_80 := "1048576"
+		query = strings.Replace(query, args_cachesize_57, args_cachesize_80, -1)
+		args_cachetype_57 := "@@query_cache_type"
+		args_cachetype_80 := "'OFF'"
+		query = strings.Replace(query, args_cachetype_57, args_cachetype_80, -1)
+		args_txiso_57 := "@@tx_isolation"
+		args_txiso_80 := "@@transaction_isolation"
+		query = strings.Replace(query, args_txiso_57, args_txiso_80, -1)
+		log.Debugf("ComQueryFor8.0: %s", query)
```
Comment (jjeffcaii) [inline pkg/runtime/plan/dml/simple_select.go:86]: Use camel style instead of underscore.

**Review by jjeffcaii — COMMENTED** [review]
pls fix.

**Comment by jjeffcaii** [comment]
PS: This is only a workaround, we should handle it by visiting AST nodes in the future.

File: pkg/runtime/plan/dml/simple_select.go (hunk 1 above)
Comment (csynineyang) [reply pkg/runtime/plan/dml/simple_select.go:86]: OK

**Comment by csynineyang** [comment]
> PS: This is only a workaround, we should handle it by visiting AST nodes in the future.

Yes，I think so 

I'am not familiar with source code very much，so I have to fix it by the simple way

**Review by binbin0325 — APPROVED** [review]

File: integration_test/scene/db_tbl_rw/integration_test.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+ */
+
+package test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/arana-db/arana/test"
+	_ "github.com/go-sql-driver/mysql"
```
Comment (baerwang) [inline integration_test/scene/db_tbl_rw/integration_test.go:25]: import style

Comment (csynineyang) [reply integration_test/scene/db_tbl_rw/integration_test.go:25]: source code was formatted by vs. code, : (

Comment (dongzl) [reply integration_test/scene/db_tbl_rw/integration_test.go:26]: > source code was formatted by vs. code, : (

Hi. @csynineyang , you can use `imports-formatter` to format import.

```
go get -u github.com/dubbogo/tools/cmd/imports-formatter

imports-formatter
```

Comment (csynineyang) [reply integration_test/scene/db_tbl_rw/integration_test.go:26]: got it

**Review by baerwang — APPROVED** [review]

**Pull request merged**